fake_index.js

```diff
@@ -0,0 +1,166 @@
+const express = require('express');
+const cors = require('cors');
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+// require your store library
+const KnexSessionStore = require('connect-session-knex')(session);
+
+const db = require('./database/dbConfig.js');
+
+const server = express();
+const sessionConfig = {
+  name: 'monkey', // default is connect.sid
+  secret: 'nobody tosses a dwarf!',
+  cookie: {
+    maxAge: 1 * 24 * 60 * 60 * 1000, // a day
+    secure: false, // only set cookies over https. Server will not send back a cookie over http.
+  }, // 1 day in milliseconds
+  httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+  resave: false,
+  saveUninitialized: false,
+  store: new KnexSessionStore({
+    tablename: 'sessions',
+    sidfieldname: 'sid',
+    knex: db,
+    createtable: true,
+    clearInterval: 1000 * 60 * 60,
+  }),
+};
+
+server.use(session(sessionConfig));
+
+server.use(express.json());
+server.use(cors());
+
+function protected(req, res, next) {
+  if (req.session && req.session.username) {
+    next();
+  } else {
+    res.status(401).json({ message: 'you shall not pass!!' });
+  }
+}
+
+server.get('/', (req, res) => {
+  res.send('Its Alive!');
+});
+
+server.post('/api/register', (req, res) => {
+  // grab credentials
+  const creds = req.body;
+
+  // hash the password
+  const hash = bcrypt.hashSync(creds.password, 10);
+
+  // a > 345 > 3er > 45d >  n times = 2^10 =
+
+  // TLS secure communication between nodes
+  // computer > isp > node1 > node 3 > node 13 > server
+
+  // replace the user password with the hash
+  creds.password = hash;
+
+  // save the user
+  db('users')
+    .insert(creds)
+    .then(ids => {
+      const id = ids[0];
+
+      // return 201
+      res.status(201).json(id);
+    })
+    .catch(err => res.status(500).send(err));
+});
+
+server.post('/api/login', (req, res) => {
+  // grab creds
+  const creds = req.body;
+
+  // find the user
+  db('users')
+    .where({ username: creds.username })
+    .first()
+    .then(user => {
+      // check creds
+      if (user && bcrypt.compareSync(creds.password, user.password)) {
+        // grab roles for user
+        // req.session.roles = roles;
+        req.session.username = user.username;
+
+        res.status(200).send(`Welcome ${req.session.username}`);
+      } else {
+        res.status(401).json({ message: 'You shall not pass!' });
+      }
+    })
+    .catch(err => res.status(500).send(err));
+});
+
+server.get('/api/logout', (req, res) => {
+  if (req.session) {
+    req.session.destroy(err => {
+      if (err) {
+        res.send('error logging out');
+      } else {
+        res.send('good bye');
+      }
+    });
+  }
+});
+
+server.get('/setname', (req, res) => {
+  req.session.name = 'Frodo';
+  res.send('got it');
+});
+
+server.get('/greet', (req, res) => {
+  const name = req.session.username;
+  res.send(`hello ${name}`);
+});
+
```

Comment:
I can't remember if we've talked about how to conceal these before, but remember that it isn't safe to push any sort of token/hashing where the public can see. Let me know if you need help concealing them.

Reply:
Referring to lines 119 && 120.

Reply:
This was a go-by I was using, it was Luis's code example from lecture. I meant to delete the file before doing any push's, but it seems as though I didn't!